include/crocoddyl/multibody/residuals/pair-collisions.hpp

```diff
@@ -0,0 +1,157 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2021, LAAS-CNRS, University of Edinburgh
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef CROCODDYL_MULTIBODY_RESIDUALS_PAIR_COLLISIONS_HPP_
+#define CROCODDYL_MULTIBODY_RESIDUALS_PAIR_COLLISIONS_HPP_
+
+#include "crocoddyl/core/residual-base.hpp"
+#include "crocoddyl/multibody/states/multibody.hpp"
+#include "crocoddyl/multibody/data/multibody.hpp"
+#include "crocoddyl/core/utils/exception.hpp"
+#include "pinocchio/multibody/geometry.hpp"
+#include "pinocchio/algorithm/jacobian.hpp"
+#include "pinocchio/algorithm/geometry.hpp"
+#include "pinocchio/multibody/fcl.hpp"
+
+namespace crocoddyl {
+
+/**
+ * @brief Pair collisions residual
+ *
+ * This residual function defines the euclidean distance between a geometric collision pair as 
+ * \f$\mathbf{r}=\mathbf{p}_1-\mathbf{p}_2^*\f$, where \f$\mathbf{p}_1,\mathbf{p}_2^*\in~\mathbb{R}^3\f$
+ * are the nearest points on each collision object with respect to the other object. One of the objects is
+ * a body frame of the robot, the other is an external object. Note that for the
+ * sake of fast computation, it is easier to define the collision objects as inflated capsules.
+ * Note also that the dimension of the residual vector is obtained from 3. Furthermore, the Jacobians 
+ * of the residual function are computed analytically.
+ *
+ * As described in `ResidualModelAbstractTpl()`, the residual value and its Jacobians are calculated by `calc` and
+ * `calcDiff`, respectively.
+ *
+ * \sa `ResidualModelAbstractTpl`, `calc()`, `calcDiff()`, `createData()`
+ */
+template <typename _Scalar>
+class ResidualModelPairCollisionsTpl : public ResidualModelAbstractTpl<_Scalar> {
```

Comment:
1. Rename this class as `ResidualModelPairCollisionTpl`. (I removed the `s`)
2. Do 1 for the Python bindings